tests/test_middleware_interface.py

```diff
@@ -362,6 +361,52 @@ def test_run_pipeline_empty_quantum_graph(self):
         with self.assertRaisesRegex(RuntimeError, "No data to process"):
             self.interface.run_pipeline(self.next_visit, {2})
 
+    def _assert_in_collection(self, butler, collection, dataset_type, data_id):
+        for dataset in butler.registry.queryDatasets(dataset_type, collections=collection, dataId=data_id):
+            return
+        self.fail(f"No datasets found matching {dataset_type}@{data_id} in {collection}.")
```

Comment:
Would this work instead of the for loop: `self.assertIn(dataset_type, list(butler.registry.queryDatasets(...)))`? `for thing: return` tripped me up.

Reply:
That requires reading in all the dataset refs, which, since this is the DECam dataset, can be sizeable.

Reply:
Yeah, I think a comment like that might help.